samcli/lib/sync/flows/rest_api_sync_flow.py

```diff
@@ -48,15 +52,157 @@ def __init__(
+                    continue
+                # If the stage's deployment ID is not static and the rest API ID matchs, then update
+                for item in get_resource_ids_by_type(self._stacks, AWS_APIGATEWAY_DEPLOYMENT):
+                    if item.logical_id == dep_id:
+                        if rest_api_id == self._api_identifier:
+                            stages.add(cast(str, stage_dict.get("Properties").get("StageName")))  # type: ignore
+
+        return stages
+
+    def _update_stages(self, stages: Set[str], deployment_id: Optional[str]) -> Set[str]:
+        """
+        Update all the relevant stages
+
+        Parameters
+        ----------
+        stages: Set[str]
+            The set of stage names to be updated
+        deployment_id: Optional[str]
+            The newly created deployment ID to be used in the stages
+        Returns
+        ----------
+        Set[str]: A set of previous deployment IDs to be cleaned up
+        """
+        prev_dep_ids = set()
+        for stage in stages:
+            # Collects previous deployment IDs to clean up
+            response_get = cast(Dict, self._api_client.get_stage(restApiId=self._api_physical_id, stageName=stage))
+            prev_dep_id = response_get.get("deploymentId")
+            if prev_dep_id:
+                prev_dep_ids.add(cast(str, prev_dep_id))
+
+            # Updates the stage with newest deployment
+            LOG.debug("%sTrying to update the stage %s through client", self.log_prefix, stage)
+            response_upd = cast(
+                Dict,
+                self._api_client.update_stage(
+                    restApiId=self._api_physical_id,
+                    stageName=stage,
+                    patchOperations=[{"op": "replace", "path": "/deploymentId", "value": deployment_id}],
+                ),
+            )
+            LOG.debug("%sUpdate Stage Result: %s", self.log_prefix, response_upd)
+
+            # Flushes the cache so that the changes are calleable
+            self._api_client.flush_stage_cache(restApiId=self._api_physical_id, stageName=stage)
+            self._api_client.flush_stage_authorizers_cache(restApiId=self._api_physical_id, stageName=stage)
+        return prev_dep_ids
+
+    def _delete_deployments(self, prev_deployment_ids: Set[str]) -> None:
+        """
+        Delete the previous deployment
+
+        Parameters
+        ----------
+        prev_deployment_ids: Set[str]
+            A set of previous deployment IDs to be cleaned up
+        """
+        for prev_dep_id in prev_deployment_ids:
+            LOG.debug("%sTrying to delete the previous deployment %s through client", self.log_prefix, prev_dep_id)
+            try:
+                response_del = cast(
+                    Dict, self._api_client.delete_deployment(restApiId=self._api_physical_id, deploymentId=prev_dep_id)
+                )
+                LOG.debug("%sDelete Deployment Result: %s", self.log_prefix, response_del)
```

Comment:
Same as above.
Can we also move `LOG.debug("%sTrying to delete the previous deployment %s through client", self.log_prefix, prev_dep_id)` into the try block so it's easier to read?